src/Hosting/TestHost/src/HttpContextBuilder.cs

```diff
@@ -171,18 +173,9 @@ private async Task CompleteRequestAsync()
                 await _requestPipe.Reader.CompleteAsync();
             }
 
-            if (_sendRequestStreamTask != null)
-            {
-                try
-                {
-                    // Ensure duplex request is either completely read or has been aborted.
-                    await _sendRequestStreamTask;
-                }
-                catch (OperationCanceledException)
-                {
-                    // Request was canceled, likely because it wasn't read before the request ended.
-                }
-            }
+            // Don't wait for request to drain. It could block indefinitely. In a real server
+            // we would wait for a timeout and then kill the socket.
+            // Potential future improvement: add logging that the request timed out
```

Comment:
I think reordering CompleteResponseAsync and CompleteRequestAsync makes some sense since the client could hypothetically be awaiting a full response before completing its request.

Not awaiting the _sendRequestStreamTask worries me a little. What if it throws a non-OperationCanceledException and we don't observe it now? I'm not sure that logging is sufficient since the test might be over by the time any error is logged if CompleteRequestAsync doesn't await it.

Other than a situation like in the ClientStreaming_ResponseCompletesWithoutResponseBodyWrite test where the request stream task is intentionally held up until the end of the test, why would awaiting the _sendRequestStreamTask be problematic?

Reply:
> Other than a situation like in the ClientStreaming_ResponseCompletesWithoutResponseBodyWrite test where the request stream task is intentionally held up until the end of the test, why would awaiting the _sendRequestStreamTask be problematic?

A delayed request body is enough, it puts you into a deadlock condition.